src/hotspot/share/gc/shenandoah/shenandoahFreeSet.hpp

```diff
@@ -146,48 +254,99 @@ class ShenandoahSetsOfFree {
   void assert_bounds() NOT_DEBUG_RETURN;
 };
 
+// Publicly, ShenandoahFreeSet represents memory that is available to mutator threads.  The public capacity(), used(),
+// and available() methods represent this public notion of memory that is under control of the mutator.  Separately,
+// ShenandoahFreeSet also represents memory available to garbage collection activities for compaction purposes.
+//
+// The Shenandoah garbage collector evacuates live objects out of specific regions that are identified as members of the
+// collection set (cset).
+//
+// The ShenandoahFreeSet endeavors to congregrate survivor objects (objects that have been evacuated at least once) at the
```

Comment:
May be "tries to colocate" is simpler than "endeavors to congregate"?

Reply:
Revised this comment.  Thanks.